FSJ_django20_project/FSJ/views.py

```diff
@@ -28,13 +28,31 @@ def home(request):
 @user_passes_test(is_FSJ_user)
 def profile(request):
     FSJ_user = get_FSJ_user(request.user.username)
-    if request.POST:
-        profile_form = FSJ_user.get_profile_form(request)
-        FSJ_user.update_FSJUser_from_profile_form(profile_form)
+    if request.method == "POST":
+        if isinstance(FSJ_user, Student):
+            profile_form = StudentRestrictedForm(request.POST, instance=FSJ_user)
+        elif isinstance(FSJ_user, Adjudicator):
+            profile_form = AdjudicatorRestrictedForm(request.POST, instance=FSJ_user)
+        elif isinstance(FSJ_user, Coordinator):
+            profile_form = CoordinatorRestrictedForm(request.POST, instance=FSJ_user)
+        if profile_form.is_valid():
+            profile_form.save()
+            return redirect('profile')
+        else:
+            raise ValueError
```

Comment:
We don't want a profile update to result in an exception being thrown. Since we know profile_form.is_valid() returns false, following the rest of this handler it gets added to back to the context and rendered again. The template then is able to show the errors associated via the {{field.errors}}

Reply:
Think I fixed this by changing indent in handler. Should I delete the raise statement anyways?

Reply:
Yes remove the raise ValueError because we don't want to raise a ValueError when profile_form.is_valid() returns false.
However you did not fix the problem by changing the indenting, but actually created a problem that didn't exist previously.
For example, with the current indenting, only coordinator restricted forms get a context at all, let alone being rendered
What we want is for the case that the request method is a POST and the form is invalid, it still gets added to the context and rendered by being on the same level as the if request method is post/else blocks, instead of being part of those blocks.